### PR TITLE
[Tools] Kotlin API codegen

### PR DESCRIPTION
**Pull request by miloserdow**
## Description

Add a script for generation of KDoc Kotlin documentation using Dokka tool. Solves #2217.

## How to test

Manually

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [X] Create pull request as draft initially, unless its complete.
- [X] Add tests to cover changes as needed.
- [X] Update documentation as needed.
- [X] If there is a related Issue, mention it in the description.

### CODE REVIEW

**Comment by miloserdow** [comment]
Left to do: modify release scripts if needed

**Comment by hewigovens** [comment]
Some screenshots would be nice, is there a way we can easily host it on web?

**Comment by Milerius** [comment]
> Some screenshots would be nice, is there a way we can easily host it on web?

It generates an archive which inside contains an index.html, screenshot/video coming

**Comment by miloserdow** [comment]
https://user-images.githubusercontent.com/3388747/197799186-6ad22ba7-142e-4cdb-be83-aab85147a1ad.mov

**Review by Milerius — COMMENTED** [review]
Let's invoke kotlin-doc script during android build, be sure to build an archive inside kotlin-doc for future upload

**Comment by Milerius** [comment]
Is it ready for review @miloserdow ?

**Comment by miloserdow** [comment]
> Is it ready for review @miloserdow ?

Yeah, forgot to set it

**Review by Milerius — DISMISSED** [review]

File: tools/android-build [hunk 1]
```diff
@@ -13,3 +13,8 @@ cp trustwalletcore/build/outputs/aar/trustwalletcore-release.aar ../build/trustw
 popd
 
 echo "Now upload build/trustwalletcore.aar to https://github.com/TrustWallet/trust-wallet-core/releases/tag/$version"
+
+echo "Building docs..."
+tools/kotlin-doc
+
+echo "Now upload Kotlin docs from build/kdoc.zip to whatever place it needs to be"
```
Comment (hewigovens) [inline tools/android-build:20]: let's add build and upload step to `tools/android-release` script.

File: tools/kotlin-doc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+https://repo1.maven.org/maven2/org/jetbrains/dokka/kotlin-analysis-intellij/1.7.20/kotlin-analysis-intellij-1.7.20.jar
+https://repo1.maven.org/maven2/org/jetbrains/dokka/kotlin-analysis-compiler/1.7.20/kotlin-analysis-compiler-1.7.20.jar
+https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-core/1.6.4/kotlinx-coroutines-core-1.6.4.jar
+https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-html-jvm/0.8.0/kotlinx-html-jvm-0.8.0.jar
+https://repo1.maven.org/maven2/org/freemarker/freemarker/2.3.31/freemarker-2.3.31.jar
+)
+
+pluginClassPath=""
+
+curl $DOKKA_CLI_JAR --output dokka.jar
```
Comment (hewigovens) [inline tools/kotlin-doc:19]: Can you move this step to `install-android-dependencies`?

File: tools/doxygen_convert_comments [hunk 3]
```diff
@@ -25,4 +42,15 @@ function swift_convert() {
...
+
+  for d in $KOTLIN_FOLDER_PATH ; do
+      process_kotlin_comments $d
+  done
+
+  rm -rf $KOTLIN_FOLDER_PATH_BAK
+}
+
 swift_convert
+kotlin_convert
```
Comment (hewigovens) [inline tools/doxygen_convert_comments:56]: add new line to the end of file

File: tools/android-build (hunk 1 above)
Comment (hewigovens) [reply tools/android-build:20]: Take a look at how to upload it: https://github.com/trustwallet/wallet-core/blob/master/tools/ios-xcframework-release#L85

File: .gitignore [hunk 4]
```diff
@@ -64,3 +64,5 @@ samples/cpp/sample
 
 *.xcuserdatad/
 
+# java binaries
+*.jar
```
Comment (catenocrypt) [inline .gitignore:68]: How come there are `.jar` files in `samples/pp/sample` ?

**Review by catenocrypt — APPROVED** [review]

**Review by Milerius — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]

**Pull request merged**